pilot/pkg/model/credentials/resource.go

```diff
@@ -76,21 +76,23 @@ func ToKubernetesGatewayResource(namespace, name string) string {
 }
 
 // ToResourceName turns a `credentialName` into a resource name used for SDS
-func ToResourceName(name string) string {
+func ToResourceName(name, proxyNamespace string) string {
 	if strings.HasPrefix(name, BuiltinGatewaySecretTypeURI) {
 		return "default"
 	}
 	if strings.HasPrefix(name, InvalidSecretTypeURI) {
 		return InvalidSecretTypeURI
 	}
+	if strings.HasPrefix(name, KubernetesSecretTypeURI) {
```

Comment:
@keithmattix @hzxuzhonghu @howardjohn only normalized kubernetes types. Based on my read, gateway secretes and configmaps already have namespaces? Let me know if that is not the case